src/python/pants/backend/jvm/tasks/coverage/manager.py

```diff
@@ -70,7 +70,8 @@ def register_junit_options(register, register_jvm_tool):
     register('--coverage', type=bool, fingerprint=True, help='Collect code coverage data.')
     register('--coverage-processor', advanced=True, fingerprint=True,
              choices=['cobertura', 'jacoco'], default='cobertura',
```

Comment:
Hm... I was thinking that the approach here would be to change the `default` to something else. Perhaps a `none` value? Or just literal `None`?

Reply:
I found that code structure very confusing going in to the prior change, and thus used the built-in facility for defaulting instead of doing it in code later. The default was in fact cobertura, but the user had no way to know via the normal option help - seems to me they should be able to know.

Reply:
Given that we actually have a `NoCoverage` implementation, a one-to-one mapping between potential option values and implementations would make the most sense to me. But inspecting the flaggedness of a value doesn't seem like a great solution.

So that probably means either having `choices=['cobertura', 'jacoco', 'none'], default='none'` or `choices=['cobertura', 'jacoco'], default=None`. If the latter is illegal (I hope it is?) then it would make the choice easier.

Reply:
Well, we also have `--coverage` which is what muddies all this up. My change here preserves prior behavior exactly with the 1 difference that the --coverage-processor option default is now indicated in help output. With `choices=['cobertura', 'jacoco', 'none']` we need to handle `--coverage --coverage-processor='none'`, With `choices=['cobertura', 'jacoco'], default=None` -we're back to the setup I found confusing and "fixed". Since the only observable differences between my change here and `choices=['cobertura', 'jacoco'], default=None` are:
1. Coder confusion over why we're not using the `default` facility
2. End user messaging

I can buy your stance if I revert to prior and ammend the help message to state the default that will be used if `--coverage` is specified and I can just look past the confusion this caused me. Let me know.

Reply:
There is another aspect here, but moot in this case, which is defaulting a None option in code later leads to 2 different cache keys for something that should map to 1. Not leading to any error, just not necessary.

Reply:
Aha. Darn. I missed that the `--coverage` flag exists. Alright. If you have an appetite for deprecating `--coverage` in favor of `choices=['cobertura', 'jacoco', 'none']` that would feel like the best solution to me. The downside would be that passing `--coverage-open` without a processor should fail (...but I believe that that falls naturally out of the `NoCoverage` null object?)

One potential real issue: I don't think that the "flaggedness" of an option is included in the cache key. So explicitly specifying the flag might not cause the cache key to change, but _would_ cause the behaviour to change.

Reply:
> The downside would be that passing --coverage-open without a processor should fail 
But that's a behavior change we cannot do without a deprecation cycle.

I think the only options that keep the current behavior, which consists of these aspects:
+ `--coverage-processor` defaults to 'cobertura' if unspecified
+ `--coverage-open` implies `--coverage`
+ `--coverage-processor` implies `--coverage`
... and thus can be implemented today without waiting for a deprecation cycle, are reverting to the previous `['cobertura', 'jacoco'], default=None` and enhancing the help message to note the default that will be picked if `--coverage` or my change.

Reply:
Let me take this issue out of the line of fire by reverting to the previous code with a better help message. I don't like it, but it allows the regression-fix to go forward and the fix for the confusing - imo - code to be done at leisure / with more discussion.